### PR TITLE
cargo-edit 0.8.0

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
Probably needs #84518.

**Comment by carlocab** [comment]
Rebased for #84518.

**Comment by carlocab** [comment]
We're now getting a different failure, which is progress:
```
==> /opt/homebrew/Cellar/cargo-edit/0.8.0/bin/cargo-add add clap@2 serde
Error: test failed
    Updating 'https://github.com/rust-lang/crates.io-index' index
Command failed due to unhandled error: Invalid manifest

Caused by: `cargo metadata` exited with an error: error: failed to parse manifest at `/private/tmp/cargo-edit-test-20210923-11561-1wed8we/demo-crate/Cargo.toml`

Caused by:
  no targets specified in the manifest
  either src/lib.rs, src/main.rs, a [lib] section, or [[bin]] section must be present
```

**Comment by Bo98** [comment]
Looks like the tests needs updating. Need to create a dummy `src/main.rs` file (or some other trickery to avoid that).

The workaround in the test can also be removed.

File: Formula/cargo-edit.rb [hunk 1]
```diff
@@ -24,17 +23,13 @@ def install
...
       (crate/"Cargo.toml").write <<~EOS
         [package]
         name = "demo-crate"
         version = "0.1.0"
       EOS
 
-      # Update the crates.io index. cargo-add doesn't currently handle this properly.
-      # https://github.com/killercup/cargo-edit/issues/420
-      # Remove this and the rust test dependency when this is fixed.
-      system "cargo", "search", "--limit=0"
```
Comment (carlocab) [inline Formula/cargo-edit.rb:36]: Can we make rust build-only now?

Comment (branchvincent) [reply Formula/cargo-edit.rb:36]: oops, thanks

**Comment by Bo98** [comment]
This invokes `cargo metadata` at runtime which needs cargo around.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**